packages/ui-components/src/components/Map/Legend/Legend.js

```diff
@@ -78,11 +78,16 @@ export const Legend = React.memo(
       // measure info for mapOverlayCode may not exist when location changes.
       const baseSerieses = baseMeasureInfo[mapOverlayCode]?.[seriesesKey] || [];
       const serieses = baseSerieses.filter(
-        ({ type, hideFromLegend, values = [] }) =>
-          ![MEASURE_TYPE_RADIUS, MEASURE_TYPE_POPUP_ONLY].includes(type) &&
-          hideFromLegend !== true &&
-          // Spetrum legend has values = []
-          (values.length === 0 || values.filter(value => !value?.hideFromLegend).length > 0),
+        ({ type, hideFromLegend, values = [], min, max, noDataColour }) => {
+          // if type is radius or popup-only, don't create a legend
+          if ([MEASURE_TYPE_RADIUS, MEASURE_TYPE_POPUP_ONLY].includes(type)) return false;
+          // if hideFromLegend is true, don't create a legend
+          if (hideFromLegend === true) return false;
+          // if type is spectrum or shaded-spectrum, only create a legend if min and max are set OR noDataColour is set. If noDataColour is not set, that means hideNullFromLegend has been set as true in the map overlay config. Spectrum legends 'values' property will always be []
+          if ([MEASURE_TYPE_SHADED_SPECTRUM, MEASURE_TYPE_SPECTRUM].includes(type))
+            return noDataColour ? true : !(min === null || max === null);
```

Comment:
Should probably check for `undefined` here too?

Can't do a regular falsey check since `min = 0` is a valid value